### PR TITLE
Adding weights and descriptions to `v3.4` from `next`

### PR DESCRIPTION
**Pull request by nate-double-u**
Updating file weights and descriptions -- copying from `next` to `v3.4`
Including new metrics files updates.

Contributes to #163 

/cc @spzala @chalin

### CODE REVIEW

**Comment by nate-double-u** [comment]
deploy preview: https://deploy-preview-168--etcd.netlify.app/docs/v3.4/

File: content/docs/v3.4/dev-guide/_index.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
 ---
 title: Developer guide
+weight: 3000
```
Comment (spzala) [inline content/docs/v3.4/dev-guide/_index.md:3]: nit, let's say "Guide for developers to work with ectd" (or "etcd guide for developers") - trying to make it more generic (cover both who are using and or providing development contribution to the etcd project") @nate-double-u - thanks!

**Review by spzala — COMMENTED** [review]
@nate-double-u thank you! I have a one inline comment, and also that let's handle the metrics content separately.

File: content/docs/v3.4/dev-guide/_index.md (hunk 1 above)
Comment (nate-double-u) [reply content/docs/v3.4/dev-guide/_index.md:3]: Would you like me to change this in `next` as well?

**Comment by nate-double-u** [comment]
> @nate-double-u thank you! I have a one inline comment, and also that let's handle the metrics content separately.

I'll take the metrics content out.

**Review by spzala — APPROVED** [review]
lgtm
thanks @nate-double-u can you please squash

**Pull request merged**